engine/src/build/config/darwin/BUILD.gn

```diff
@@ -38,15 +38,15 @@ config("compiler") {
   } else {
     _triplet_environment = ""
   }
-  # OS.
+  # Deployment target (OS and version).
   if (is_ios) {
-    _triplet_os = "apple-ios"
-  } else if (is_mac) {
-    _triplet_os = "apple-macos"
+    _triplet_os = "apple-ios$ios_deployment_target"
```

Comment:
I noticed there's also `ios_testing_deployment_target`. Not sure how best to determine if the current target is `testonly`.